nixos/tests/all-tests.nix

```diff
@@ -287,6 +287,7 @@ in
   man = handleTest ./man.nix {};
   mariadb-galera = handleTest ./mysql/mariadb-galera.nix {};
   mastodon = handleTestOn ["x86_64-linux" "i686-linux" "aarch64-linux"] ./web-apps/mastodon.nix {};
+  mastodon-extdb = handleTestOn ["x86_64-linux" "i686-linux" "aarch64-linux"] ./web-apps/mastodon-extdb.nix {};
```

Comment:
This shouldn't be a separate test, but rather an addition to the one we have. Too much duplicate code otherwise.

We should abstract over running a test against a mastodon instance, e.g. by introducing a function `make-mastodon-test`, that accepts `nodes`. See `nixos/tests/postgresql.nix` for inspiration, even though there it takes the postgresql version as an argument.

Also perspectively, but that is outside the scope of this PR, I think it would be cool if we'd test that all components that one can run externally (postgresql, nginx, redis, smtp) are also tested. What do you think about that?

Reply:
> This shouldn't be a separate test, but rather an addition to the one we have. Too much duplicate code otherwise.
> 
> We should abstract over running a test against a mastodon instance, e.g. by introducing a function `make-mastodon-test`, that accepts `nodes`. See `nixos/tests/postgresql.nix` for inspiration, even though there it takes the postgresql version as an argument.

Now I don't know how to implement such a test.



> Also perspectively, but that is outside the scope of this PR, I think it would be cool if we'd test that all components that one can run externally (postgresql, nginx, redis, smtp) are also tested. What do you think about that?

Running nginx from other server doesn't make sense

By default, only running `mastodon` test.
Test `mastodon-extdb` serves as an additional check.
For example - tests of nginx, systemd, mysql, run independently of each other.

Reply:
Both would be fine for me!

Reply:
Now tested this option, the presence of nginx on the server with mastodon is mandatory. I don't see the point of separating nginx in the test.

Reply:
No full checking this variant.

Reply:
Created!